test/reg-test/fam-api-reg/fam_dataitem_interleave_test.cpp

```diff
@@ -0,0 +1,415 @@
+           INTERLEAVE_SIZE);
+    const char *srcRegionName = get_uniq_str("Src_Region", my_fam);
+    const char *srcItemName = get_uniq_str("Src_Itemt", my_fam);
+
+    const char *destRegionName = get_uniq_str("Dest_Region", my_fam);
+    const char *destItemName = get_uniq_str("Dest_Itemt", my_fam);
+
+    EXPECT_NO_THROW(srcDesc = my_fam->fam_create_region(
+                        srcRegionName, 104857600, 0777, NULL));
+    EXPECT_NE((void *)NULL, srcDesc);
+
+    // Allocating data items in the created region
+    EXPECT_NO_THROW(
+        srcItem = my_fam->fam_allocate(srcItemName, 16777216, 0777, srcDesc));
+    EXPECT_NE((void *)NULL, srcItem);
+
+    EXPECT_NO_THROW(destDesc = my_fam->fam_create_region(
+                        destRegionName, 536870912, 0777, NULL));
+    EXPECT_NE((void *)NULL, destDesc);
+
+    for (int i = 0; i < ITERATION; i++) {
+        // Allocating data items in the created region
+        char itemInfo[255];
+        sprintf(itemInfo, "%s_%d", destItemName, i);
+        EXPECT_NO_THROW(destItem[i] = my_fam->fam_allocate(itemInfo, 16777216,
+                                                           0777, destDesc));
+        EXPECT_NE((void *)NULL, destItem[i]);
+    }
+    EXPECT_NO_THROW(
+        my_fam->fam_put_blocking(local, srcItem, 4096, 6 * INTERLEAVE_SIZE));
+
+    void *waitObj[ITERATION];
+    for (int i = 0; i < ITERATION; i++) {
+        EXPECT_NO_THROW(waitObj[i] =
+                            my_fam->fam_copy(srcItem, 4096, destItem[i], 1024,
+                                             6 * INTERLEAVE_SIZE));
+        EXPECT_NE((void *)NULL, waitObj[i]);
+    }
+
+    for (int i = ITERATION - 1; i >= 0; i--) {
+        EXPECT_NO_THROW(my_fam->fam_copy_wait(waitObj[i]));
+    }
+
+    // allocate local memory to receive 20 elements
+    char *local2 = (char *)malloc(6 * INTERLEAVE_SIZE);
+
+    for (int i = 0; i < ITERATION; i++) {
+        EXPECT_NO_THROW(my_fam->fam_get_blocking(local2, destItem[i], 1024,
+                                                 6 * INTERLEAVE_SIZE));
+        EXPECT_STREQ(local, local2);
+    }
+
+    EXPECT_NO_THROW(my_fam->fam_deallocate(srcItem));
+    EXPECT_NO_THROW(my_fam->fam_destroy_region(srcDesc));
+
+    for (int i = 0; i < ITERATION; i++) {
+        EXPECT_NO_THROW(my_fam->fam_deallocate(destItem[i]));
+    }
+    EXPECT_NO_THROW(my_fam->fam_destroy_region(destDesc));
+
+    delete srcItem;
+    delete srcDesc;
+    delete destDesc;
+
+    free((void *)srcRegionName);
+    free((void *)srcItemName);
+    free((void *)destRegionName);
+    free((void *)destItemName);
+}
+
+int main(int argc, char **argv) {
+    int ret = 0;
+    ::testing::InitGoogleTest(&argc, argv);
+
+    my_fam = new fam();
+
+    init_fam_options(&fam_opts);
+
+    EXPECT_NO_THROW(my_fam->fam_initialize("default", &fam_opts));
+
+    ret = RUN_ALL_TESTS();
+
+    EXPECT_NO_THROW(my_fam->fam_finalize("default"));
+
+    return ret;
+}
```

Comment:
Add couple of atomic tests to operate on different offsets (offsets can be in second stripe)